docs/developer-docs/backend/rust/9-counter.md

```diff
@@ -0,0 +1,216 @@
+# 9: Incrementing a counter
+
+## Overview
+
+In this guide, you are going to write a dapp that provides a few basic functions to increment a counter and illustrates the persistence of a value.
+
+For this guide, the dapp declares a `COUNTER` as a mutable variable to contain a natural number that represents the current value of the counter. This dapp supports the following functions:
+
+-   The `increment` function updates the current value, incrementing by 1 with no return value.
+
+-   The `get` function is a simple query that returns the current value of the counter.
+
+-   The `set` function updates the current value to the numeric value you specify as an argument.
+
+This guide provides a simple example of how you can increment a counter by calling functions on a deployed canister. By calling the function to increment a value multiple times, you can verify that the variable state—that is, the value of the variable between calls—persists.
+
+## Prerequisites
+
+Before getting started, assure you have set up your developer environment according to the instructions in the [developer environment guide](./3-dev-env.md).
+
+## Create the counter dapp project
+
+Open a terminal window on your local computer, if you don’t already have one open.
+
+Start by creating a new project by running the following command:
+
+``` bash
+dfx new --type=rust rust_counter
+```
+
+Then, navigate into your project directory by running the command:
+
+``` bash
+cd rust_counter
+```
+
+:::info
+Note: the following steps assume the terminal is still open and the current directory is `rust_counter`.
+:::
+
+Now that you have the files in place for your Rust dapp, we can replace the template `lib.rs` dapp code with some code that creates a counter dapp instead. 
+
+To replace the default dapp, open the template `src/rust_counter_backend/src/lib.rs` file in a text editor and delete the existing content.
+
+Then, copy and paste this code into the `src/rust_counter_backend/src/lib.rs` file.
+
+```rust
+use ic_cdk::export::candid;
+use ic_cdk_macros::*;
+
+static mut COUNTER: Option<candid::Nat> = None;
```

Comment:
Per above, we should not be using `static mut` or `unsafe` anywhere in the docs. This should be `thread_local!` with a `Cell`. See the counter example in the examples repo.